index.html

```diff
@@ -43,22 +49,33 @@ <h1>Pick a theme</h1>
         localStorage.setItem('selected-theme', theme);
       }
     }
+    
+    const setDefaultTheme = () => {
+      themes.default = (prefersColorMediaQuery.matches === true) ? themes.dark : themes.light
+    }
 
-    const setInitialTheme = () => {
-      const savedTheme = localStorage.getItem('selected-theme');
-      if(savedTheme && savedTheme !== defaultTheme) {
-        applyTheme(savedTheme);
+    const applyInitialTheme = () => {
+      if(themes.saved && themes.saved !== themes.default) {
+        applyTheme(themes.saved);
+        return;
       }
+      applyTheme(themes.default);
     };
 
-    setInitialTheme();
+    setDefaultTheme();
+    applyInitialTheme();
 
     const themeSwitcher = document.querySelector('.theme-switcher');
     const buttons = themeSwitcher.querySelectorAll('button');
 
     buttons.forEach((button) => {
       button.addEventListener('click', handleThemeSelection);
     });
+    
+    prefersColorMediaQuery.addEventListener('change', () => {
+      setDefaultTheme();
+      applyInitialTheme();
+    });
```

Comment:
I feel like this and line 65+66 can be combined somehow. But not sure where to go next tbh.

Reply:
I think so too 🤔 I need to sleep on it and revisit it with fresh eyes tomorrow, but I think something along these lines inside the `applyInitialTheme` function could maybe simplify it:
- If a theme is saved in the local storage: apply the saved theme.
- Otherwise, if `prefers-color-scheme` is supported: apply `themes.dark` or `themes.light` based on the media query.
- If the `prefers-color-scheme` is not supported: do nothing (it will apply based on the `:root`).